### PR TITLE
DOC: Reorganize examples

### LINKED ISSUE #566: Improve navigating the examples
Currently the examples README.txt  file has a link to the Discussions area. I think the formatting is wrong as the hyperlink returns a 404 error. But I think we should update the README.txt file to more of the heavy lifting to point the user to the correct example than pointing the user to the Discussion area. We may need to rename some of the examples as everything starts with plot_* but not every example generates a plot.

Also, recent PR #563 created a new example to show how to downsample the data before plotting to speed up 2D plotting. I would not know which example to look in to find this information from the name of the file, plot_examples.py. We should create a TOC or some other listing of the problem/example and hyperlink to that example in the examples/README.txt

### PR DESCRIPTION
**Pull request by mgrover1**
Closes #566

### CODE REVIEW

**Comment by mgrover1** [comment]
@AdamTheisen - @zssherman and I sat down and reorganized the examples... let me know what you think!

**Review by zssherman — APPROVED** [review]

**Review by AdamTheisen — COMMENTED** [review]
I know the plot_qc_example.py example shows QC data but it's really intended to show the visualization of QC information so I think it would be better suited under plotting but open to feedback here.

File: examples/qc/plot_state_variable.py [hunk 1]
```diff
@@ -9,10 +9,11 @@
 
 """
 
+from matplotlib import pyplot as plt
```
Comment (AdamTheisen) [inline examples/qc/plot_state_variable.py:12]: I think this one should be in plotting as well

File: examples/retrievals/plot_skewt.py [hunk 2]
```diff
@@ -8,9 +8,9 @@
 """
```
Comment (AdamTheisen) [inline examples/retrievals/plot_skewt.py:10]: Can we have this in plotting as well?  We do some retrievals here but not sure it's the focus of it.

**Comment by AdamTheisen** [comment]
@mgrover1 @zssherman looks good!  I like the addition of a workflows area!  I just had a couple thoughts on where some of these might be better suited but open to thoughts/feedback.

**Comment by mgrover1** [comment]
Thanks for the suggestions @AdamTheisen ! I made those changes.

**Review by AdamTheisen — APPROVED** [review]

**Comment by mgrover1** [comment]
Once the tests pass, I will merge @AdamTheisen @zssherman

**Pull request merged**